File: engines/polyglot-route-engine/fixtures/module/typescript/equivalence_module.ts
export function clampNumber(value: number, minimum: number, maximum: number): number {
  if (value < minimum) { return minimum; }
  if (value > maximum) { return maximum; }
  return value;
}

export function lowerNumber(value: number, minimum: number): number {
  if (value < minimum) { return minimum; }
  return value;
}

export function upperNumber(value: number, maximum: number): number {
  if (value > maximum) { return maximum; }
  return value;
}

export function sameString(left: string, right: string): boolean {
  return left === right;
}

export function concatString(left: string, right: string): string {
  return left + right;
}

export function both(left: boolean, right: boolean): boolean {
  return left && right;
}

File: engines/polyglot-route-engine/fixtures/module/csharp/EquivalenceModule.cs
public static class EquivalenceModule
{
    public static long calculate(long subtotal, long tax)
    {
        if (subtotal < 0) { return 0; }
        return subtotal + tax;
    }

    public static long clamp(long value, long minimum, long maximum)
    {
        if (value < minimum) { return minimum; }
        if (value > maximum) { return maximum; }
        return value;
    }

    public static long difference(long left, long right) => left - right;

    public static double clampNumber(double value, double minimum, double maximum)
    {
        if (value < minimum) { return minimum; }
        if (value > maximum) { return maximum; }
        return value;
    }

    public static bool both(bool left, bool right) => left && right;
}
